### DISCUSSION TITLE
[Bug]: The webviewerloaded event is not fired in iframe

### DISCUSSION THREAD

**Discussion by uforulez** [mozilla/pdf.js#19188] [Q&A]
### Attach (recommended) or Link to PDF file

https://doksi.net/en/get.php?lid=40596

### Web browser and its version

Firefox 133.0, Edge 131.0

### Operating system and its version

Windows 11

### PDF.js version

pdfjs-4.8.69-legacy-dist

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

The following events are fired when PDF.js is embedded in the parent page, but not fired when running in an iframe.

document.addEventListener("webviewerloaded", () => {
        PDFViewerApplication.initializedPromise.then(() => {

          PDFViewerApplication.eventBus.on("documentloaded", (e) => {
            console.log("documentloaded");
          });

          PDFViewerApplication.eventBus.on("scalechanging", (e) => {
            console.log("scalechanging");
          });
        });
      });

### What is the expected behavior?

I want to capture these events to track them in my Analytics software, so these events should be fired. The iframe and the parent page are in the same origin. 

### What went wrong?

Described above.

**Comment by jonasmusall** [comment]
This is because it is fired on the parent document: https://github.com/mozilla/pdf.js/blob/129fc66f65fd4fb63cb08b8c5a86ad60dc12336b/web/viewer.js#L242-L253

So you have to listen for it on the parent document as well. Either move your logic there (however then you can't access objects in the `iframe`), or use

```js
parent.document.addEventListener("webviewerloaded", () => ...)
```
